### ISSUE TITLE
EagleSoft Patient Tables

### ISSUE THREAD

**Issue by knappkl** [wdduncan/NDPBRN#8]
Do these tables need to be re-run once we are comfortable with the Tooth History tables?

**Comment by wdduncan** [comment]
This is for the patient and provider tables. We are comfortable with the tooth history tables.

**Comment by knappkl** [comment]
Bill will provide query for the Patient tables. No need to re-run Provider tables.

**Comment by wdduncan** [comment]
Please assign this to Jim. He know where the query is for extracting patients from the patients table in Eaglesoft.

**Comment by knappkl** [comment]
@jimjzheng Here is the query that will extract patients from the patient table:

SELECT DISTINCT
  patient_id, sex, birth_date, status, practice_id as patient_db_practice_id 
FROM patient
WHERE patient_id IN
 (
  SELECT DISTINCT
    txd.patient_id
  FROM
    transactions_detail txd
  WHERE txd.tran_num IN
    (
     SELECT txh.tran_num 
     FROM transactions_header txh
     WHERE txh.type = 'S'  -- filter on type is 'S'ervice
     AND txh.status = 'A'  -- filter on status is 'A'ctive' transactions (not deleted)
    )
  )

The output of this query needs to be saved as a tab delimited file include the column headers. Chen will translate the data from this file into the ontology.

Basically, the query only selects those patients that have had a service performed by the provider. Since Ted has already filtered the data for us, we don’t need the filtering criteria that was in the old query.

Please review the query and forward to Ted.
